addons/product/models/product_catalog_mixin.py

```diff
@@ -128,6 +128,7 @@ def _get_action_add_from_catalog_extra_context(self):
         return {
             'display_uom': self.env.user.has_group('uom.group_uom'),
             'product_catalog_order_id': self.id,
+            'product_catalog_order_state': self.state if "state" in self._fields else None,
```

Comment:
I don't understand why we need that here since it's only used in `purchase_stock` and you already added `po_state` in the `purchase_stock` `_get_action_add_from_catalog_extra_context` override.
```suggestion
```
I moved this line into the `purchase_stock` override and removed the `po_state` key (which seems to not be used anyway)